infrastructure/file_settings_source_test.go

```diff
@@ -0,0 +1,82 @@
+package infrastructure_test
+
+import (
+	"encoding/json"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	. "github.com/cloudfoundry/bosh-agent/infrastructure"
+
+	boshlog "github.com/cloudfoundry/bosh-agent/logger"
+	boshsettings "github.com/cloudfoundry/bosh-agent/settings"
+	fakesys "github.com/cloudfoundry/bosh-agent/system/fakes"
+)
+
+var _ = Describe("FileSettingsSource", func() {
+	var (
+		fs     *fakesys.FakeFileSystem
+		source *FileSettingsSource
+		logger boshlog.Logger
+	)
+
+	BeforeEach(func() {
+		fs = fakesys.NewFakeFileSystem()
+		logger = boshlog.NewLogger(boshlog.LevelNone)
+	})
+
+	Describe("PublicSSHKeyForUsername", func() {
+		It("returns an empty string", func() {
+			publicKey, err := source.PublicSSHKeyForUsername("fake-username")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(publicKey).To(Equal(""))
+		})
+	})
+
+	Describe("Settings", func() {
+		Context("when the settings file exists", func() {
```

Comment:
Thank you for your note. I used [file metadata service test](https://github.com/cloudfoundry/bosh-agent/blob/master/infrastructure/file_metadata_service_test.go#L32-L50) as an example for this test. I added test for invalid format both to file_settings_source_test.go and file_metadata_service_test.go.